collect_app/src/test/java/org/odk/collect/android/entities/EntitiesRepositoryTest.kt

```diff
@@ -879,6 +879,56 @@ abstract class EntitiesRepositoryTest {
         val repository = buildSubject()
         repository.save("wines", Entity.New("1", "Léoville Barton 2008",))
 
-        repository.query("wines", Query.Eq("score", "92"))
+        repository.query("wines", Query.StringEq("score", "92"))
+    }
+
+    @Test
+    fun `#query returns matching entities with numeric eq selection arguments`() {
```

Comment:
I'd like to split these 2 tests into 4 so we have individual ones for int and decimal property values.

Reply:
It won't work:
1. `Query.NumericEq` and `Query.NumericNotEq` take a double, so I always have to put a double there,e no matter if I want to test integers or doubles.
2. In properties, it's also worth to mix integers and doubles to make sure eg, the ones with `5` will be returned if I have `5.0` in the filter.